be/src/block_cache/starcache_wrapper.cpp

```diff
@@ -46,11 +48,25 @@ Status StarCacheWrapper::write_buffer(const std::string& key, const IOBuffer& bu
     if (!options) {
         return to_status(_cache->set(key, buffer.const_raw_buf(), nullptr));
     }
+
     starcache::WriteOptions opts;
     opts.ttl_seconds = options->ttl_seconds;
     opts.overwrite = options->overwrite;
-    auto st = to_status(_cache->set(key, buffer.const_raw_buf(), &opts));
-    if (st.ok()) {
+    opts.async = options->async;
+    opts.keep_alive = options->keep_alive;
+    opts.callback = options->callback;
+    opts.mode = _enable_through_data_mode ? starcache::WriteOptions::WriteMode::WRITE_THROUGH
+                                          : starcache::WriteOptions::WriteMode::WRITE_BACK;
+    Status st;
+    {
+        // The memory when writing starcache is no longer recorded to the query memory.
+        // Because we free the memory in other threads in starcache library, which is hard to track.
+        // It is safe because we limit the flying memory in starcache, also, this behavior
+        // doesn't affect the process memory tracker.
+        SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(nullptr);
```

Comment:
> do you have any idea how to trace that part memory? if it's not in query memory, then it's ony in process memory. which is hard to reason where memory goes to.

I think it is difficult to trace the memory which are delay freed in other threads of a thirdparty library using our mem tracker. We cannot make sure all memory can be traced by query memtracker, that's why we use the `SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(nullptr)` in many cases in our BE to handle these special cases. However, I don't think this part will use too much memory which not be traced, because we only limit a very small flying memory (default is 2M) when population the starcache.